### PR TITLE
feat: add lockFactory config

### PR DESCRIPTION
**Pull request by sublimator**
We are using SuperTokens with an extension that needs to refresh the tokens for a given domain that may be very infrequently browsed, or on the other hand may be browsed concurrently with the extension usage. 

We want to use supertokens-website from the background page of the extension, overriding the CookieHandler (to set tokens on the app domain using Web Extension apis) and customizing browser-tabs-tokens so that the extension can act as "just another tab" 

Wanted to get a feel if this PR is moving in the right direction and would have a chance to be merged. 

Thanks!

### CODE REVIEW

**Comment by rishabhpoddar** [comment]
Thanks for this PR @sublimator .

@porcellus will review this in the coming days.

File: lib/ts/utils/windowHandler/index.ts
Comment (porcellus) [inline lib/ts/utils/windowHandler/index.ts:24]: We shouldn't limit this to a subset. This lib exports the window handler, and the other props are used by supertokens-web-js and supertokens-auth-react.

File: lib/ts/utils/windowHandler/index.ts
Comment (porcellus) [inline lib/ts/utils/windowHandler/index.ts:31]: These updates do not match the coding style used elsewhere in the project; please run `npm run build-pretty`.
Also, I'm not sure if you've installed the commit hooks (check [here](https://github.com/supertokens/supertokens-website/blob/master/CONTRIBUTING.md#development-setup) if not)

File: lib/ts/types.ts
Comment (porcellus) [inline lib/ts/types.ts:47]:
```suggestion
     * This allows for a Lock factory to be configured, which defaults to browser-tabs-lock.
     * This can be used, for example, by a WebExtension that needs to update cookies for
     * a domain that may or may not have an associated tab open.
```

**Review by porcellus — CHANGES_REQUESTED** [review]
This is a good change; there are just a few minor issues :)

File: lib/ts/utils/windowHandler/index.ts
Comment (sublimator) [reply lib/ts/utils/windowHandler/index.ts:24]: Fair, I shouldn't have committed this

File: lib/ts/utils/windowHandler/index.ts
Comment (sublimator) [reply lib/ts/utils/windowHandler/index.ts:31]: image

I think prettier might need to be updated ? Seems this is using 1.x while 2.x is out.

**Comment by sublimator** [comment]
@porcellus 

I removed the limiting of the windowHandler type, ran prettier, updated it when it wasn't working with some of the syntax I used (`??` operator and `import type`), then tweaked the .prettierrc to be as close to 1.x as possible.   

If you are the maintainer (or one of), feel free to take over PR if it's more convenient for you.  

We just need some ability to customize the lock.

Thanks!

**Comment by sublimator** [comment]
I made the lockFactory async. I'm not sure if we will need it, but it's more flexible, so why not?

**Review by porcellus — APPROVED** [review]
LGTM :)

File: package.json
Comment (rishabhpoddar) [inline package.json:20]: - Please update the changelog under the [unreleased] section
- Might want to expose the LockFactory in the `/utils/lockFactory` folder similar to how the [windowHandler is exposed](https://github.com/sublimator/supertokens-website/blob/nd-add-lock-factory-to-config-2023-01-13/utils/windowHandler/index.js)

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: package.json
Comment (rishabhpoddar) [reply package.json:20]: The other changes look good. So once these two comments are addressed, i'll merge and release this.

**Comment by sublimator** [comment]
@rishabhpoddar 

I'm not sure how/why I set the target branch to 15.0 and not master ?
Is that right? 

Anyway, I (hopefully) addressed your two comments.

**Review by rishabhpoddar — APPROVED** [review]

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**
